dbt/include/spark/macros/materializations/incremental/incremental.sql

```diff
@@ -26,9 +26,11 @@
 
   {{ run_hooks(pre_hooks) }}
 
+  {% set is_delta = (file_format == 'delta' and existing_relation.is_delta) %}
+
   {% if existing_relation is none %}
     {% set build_sql = create_table_as(False, target_relation, sql) %}
-  {% elif existing_relation.is_view or full_refresh_mode %}
+  {% elif existing_relation.is_view or (full_refresh_mode and not is_delta) %}
     {% do adapter.drop_relation(existing_relation) %}
     {% set build_sql = create_table_as(False, target_relation, sql) %}
   {% else %}
```

Comment:
I think we still want to run `create_table_as` (which dbt will template out to `create or replace` for Delta tables) — we just want to skip the `drop` step
```suggestion
  {% elif existing_relation.is_view or full_refresh_mode %}
    {% if not is_delta %} {#-- If Delta, we will `create or replace` below, so no need to drop --#}
      {% do adapter.drop_relation(existing_relation) %}
    {% endif %}
    {% set build_sql = create_table_as(False, target_relation, sql) %}
  {% else %}
```

Reply:
@jtcohen6 Thanks, that's an important fix of the code. Committed your suggested changes directly.
Should hopefully be good to go now?